streams/src/main/java/org/apache/kafka/streams/state/internals/HasNextCondition.java

```diff
@@ -16,10 +16,8 @@
  */
 package org.apache.kafka.streams.state.internals;
 
-import org.apache.kafka.streams.KeyValue;
-import org.apache.kafka.streams.state.WindowStoreIterator;
+import org.apache.kafka.streams.state.KeyValueIterator;
 
-public interface PeekingWindowIterator<V> extends WindowStoreIterator<V> {
-
-    KeyValue<Long, V> peekNext();
+interface HasNextCondition<K> {
```

Comment:
This interface is a bit awkward to me: in session windows it is typed `Bytes` while in timed windows it is typed `Long`: also I am not sure if typed `Long` is correct since the passed in iterator is always in `<Bytes, byte[]>`.

I'm feeling it is not worthwhile to make it an interface and we could probably just implement the specific hasNext inlined in these two classes?